udata_hydra/app.py

```diff
@@ -179,21 +158,61 @@ async def get_check(request: web.Request) -> web.Response:
 @routes.get("/api/checks/all/")
 async def get_checks(request: web.Request) -> web.Response:
     url, resource_id = _get_args(request)
-    column = "url" if url else "resource_id"
-    q = f"""
-    SELECT catalog.id as catalog_id, checks.id as check_id,
-           catalog.status as catalog_status, checks.status as check_status, *
-    FROM checks, catalog
-    WHERE catalog.{column} = $1
-    AND catalog.url = checks.url
-    ORDER BY created_at DESC
-    """
-    data = await request.app["pool"].fetch(q, url or resource_id)
+    data = await Check.get_all(url, resource_id)
     if not data:
         raise web.HTTPNotFound()
     return web.json_response([CheckSchema().dump(dict(r)) for r in data])
 
 
+@routes.post("/api/checks/create/")
+async def create_check(request: web.Request) -> web.Response:
+    """
+    Force a new crawl for a given URL or resource_id, create a new corresponding check and update the catalog table with the new check id
+    """
+    try:
+        payload = await request.json()
+        url: str = payload.get("url", None)
```

Comment:
Nit: should the type be `str | None` here?

Reply:
Indeed, fixed